### PR TITLE
Add Decor3D, ThumbMarkerDecorSPace, ColorDecor3D

### PR DESCRIPTION
**Pull request by srd-software**
Add Decor3D, ThumbMarkerDecorSPace, ColorDecor3D

### CODE REVIEW

File: atflatcontrols/atscrollbar.pas [hunk 1]
```diff
@@ -850,36 +854,76 @@ procedure TATScrollbar.DoPaintStd_Thumb(C: TCanvas; const R: TRect);
   NOffset:= FTheme^.ThumbMarkerOffset;
 
   P:= CenterPoint(R);
+  if FTheme^.Decor3D then
+  begin
+    if IsHorz then P.X := p.X + 1;
+    if not IsHorz then P.Y := p.Y + 1;
```
Comment (Alexey-T) [inline atflatcontrols/atscrollbar.pas]: if IsHorz then Inc(P.X) else Inc(P.Y);

File: atflatcontrols/atscrollbar.pas [hunk 2]
```diff
@@ -544,15 +549,15 @@ procedure TATScrollbar.MouseDown(Button: TMouseButton; Shift: TShiftState;
...
+
+        if FLargeChange > 0 then
+          ScrollVal := FLargeChange; //optional
 
         if FMouseDownOnPageUp then
-          DoScrollBy(-ScrollVal)
+            DoScrollBy(-ScrollVal)
         else
-          DoScrollBy(ScrollVal);
+            DoScrollBy(ScrollVal);
```
Comment (Alexey-T) [inline atflatcontrols/atscrollbar.pas]: restore my change

File: atflatcontrols/atscrollbar.pas [hunk 3]
```diff
@@ -544,15 +549,15 @@ procedure TATScrollbar.MouseDown(Button: TMouseButton; Shift: TShiftState;
...
       else
       begin
-        if FLargeChange>0 then
-          ScrollVal:= FLargeChange
-        else
-          ScrollVal:= FPageSize;
+        ScrollVal := FPageSize; //default
+
+        if FLargeChange > 0 then
+          ScrollVal := FLargeChange; //optional
```
Comment (Alexey-T) [inline atflatcontrols/atscrollbar.pas:556]: restore my change

File: atflatcontrols/atscrollbar.pas [hunk 4]
```diff
@@ -850,36 +854,76 @@ procedure TATScrollbar.DoPaintStd_Thumb(C: TCanvas; const R: TRect);
...
-        C.LineTo(P.X-2*i, R.Bottom-NOffset);
+        C.MoveTo(P.X-DecorSpace*i, R.Top+NOffset);
+        C.LineTo(P.X-DecorSpace*i, R.Bottom-NOffset);
         if i>0 then
         begin
-          C.MoveTo(P.X+2*i, R.Top+NOffset);
-          C.LineTo(P.X+2*i, R.Bottom-NOffset);
+          C.MoveTo(P.X-DecorSpace*i, R.Top+NOffset);
+          C.LineTo(P.X-DecorSpace*i, R.Bottom-NOffset);
         end;
```
Comment (Alexey-T) [inline atflatcontrols/atscrollbar.pas:876]: changed + to - ?? this block painted paired (mirror) lines of gripper.

File: atflatcontrols/atscrollbar.pas [hunk 5]
```diff
@@ -850,36 +854,76 @@ procedure TATScrollbar.DoPaintStd_Thumb(C: TCanvas; const R: TRect);
...
+      C.Pen.Color:= ColorToRGB(FTheme^.ColorDecor3D);
+      for i:= 0 to FTheme^.ThumbMarkerDecorSize-1 do
+      begin
+        C.MoveTo((P.X-DecorSpace*i) -1, R.Top+NOffset);
+        C.LineTo((P.X-DecorSpace*i) -1, R.Bottom-NOffset);
+        if i>0 then
+        begin
+          C.MoveTo((P.X-DecorSpace*i) -1, R.Top+NOffset);
+          C.LineTo((P.X-DecorSpace*i) -1, R.Bottom-NOffset);
+        end;
```
Comment (Alexey-T) [inline atflatcontrols/atscrollbar.pas:891]: changed + to - ?

**Review by Alexey-T — CHANGES_REQUESTED** [review]

File: atflatcontrols/atscrollbar.pas (hunk 5 above)
Comment (srd-software) [reply atflatcontrols/atscrollbar.pas:891]: Yeah, got carried away in copy and paste. Sending new file now

**Comment by srd-software** [comment]
Let me know if you want a new pull request! Not sure if all these fixes will make it OK. Just let me know.

File: atflatcontrols/atscrollbar.pas (hunk 4 above)
Comment (srd-software) [reply atflatcontrols/atscrollbar.pas:876]: THANKS for spotting this. I was wondering why I didn't have the right number of grippys on the bar...

File: atflatcontrols/atscrollbar.pas (hunk 2 above)
Comment (srd-software) [reply atflatcontrols/atscrollbar.pas]: I think I got this one, last fix sent (spacing fix in code)

File: atflatcontrols/atscrollbar.pas (hunk 3 above)
Comment (srd-software) [reply atflatcontrols/atscrollbar.pas:556]: Should be OK now

File: atflatcontrols/atscrollbar.pas (hunk 1 above)
Comment (srd-software) [reply atflatcontrols/atscrollbar.pas]: Should be OK now

**Pull request merged**